### PR TITLE
proto3 switch-over RFC

### PR DESCRIPTION
**Pull request by cschuet**
[Rendered](https://github.com/cschuet/rfcs/blob/proto3/text/0000-proto3-transition.md)

### CODE REVIEW

**Review by wohe — APPROVED** [review]

**Comment by SirVer** [comment]
Should the plan for `cartographer_ros` to become a thin wrapper using the gRPC based cores be documented in this RFC as well?

**Comment by cschuet** [comment]
I think that's a great idea but should be documented in a different RFC.

**Review by SirVer — APPROVED** [review]

File: text/0000-proto3-transition.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+- We add a script 'install_proto3.sh' which clones proto3 from HEAD, compiles and installs it to '/usr/local'.
+- This script is used in our Dockerfiles to install proto3 and replaces our former 'apt install protobuf-compiler' which installed proto2.
+- We update the build instructions for Cartographer to include proto3 as a necessary prerequisite.
+
+A similar PR will be written for the 'cartographer_ros' repo.
+
+## Discussion Points
+[discussion]: #discussion
+
+[FirstPR]: https://github.com/googlecartographer/cartographer/pull/635
```
Comment (gaschler) [inline text/0000-proto3-transition.md]: I think it should be https://github.com/googlecartographer/cartographer/pull/644 instead

**Review by gaschler — APPROVED** [review]
Looks good, one comment

File: text/0000-proto3-transition.md (hunk 1 above)
Comment (cschuet) [reply text/0000-proto3-transition.md]: Done.

File: text/0000-proto3-transition.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+## Approach
+[approach]: #approach
+
+Fortunately the proto3 library is backwards compatible in the sense that it can compile proto2 messages.
+We can therefore perform the switchover in two steps:
+- Upgrade the Cartographer Protocol Buffer dependency from 2 to 3.
+- Refactor any code that uses proto2 features ('has...') and convert (at the very least) those messages that we intend to send over the wire to proto3 style.
+
+This [PR][FirstPR] summarizes the plan for step 1 for the Cartographer repo:
+- We add a script 'install_proto3.sh' which clones proto3 from HEAD, compiles and installs it to '/usr/local'.
```
Comment (ojura) [inline text/0000-proto3-transition.md:27]: From `HEAD`, or from a certain version, like Ceres?

Comment (cschuet) [reply text/0000-proto3-transition.md:27]: Good call. I think Holger already has a follow-up PR where he does exactly that.

**Pull request merged**